doc/platform-optimization/README.md

```diff
@@ -0,0 +1,188 @@
+    name: default
+    namespaces:
+    - "*"
+    shareIdleCPUsInSame: numa
+  instrumentation:
+    httpEndpoint: :8891
+    prometheusExport: true
+    reportPeriod: 60s
+    samplingRatePerMillion: 0
+  log:
+    source: true
+    debug: ["policy"]
+  pinCPU: true
+  pinMemory: false
+  reservedPoolNamespaces:
+  - kube-system
+  reservedResources:
+    cpu: "2"
+```
+
+The balloons policy creates "balloons" of CPUs that only containers
+assigned into a balloon are allowed to use.
+
+The most important options in the above configuration example are:
+
+- `allocatorTopologyBalancing: true`. This option ensures that
+  balloons (sets of allowed CPUs) are balanced between CPU sockets in
+  the system, and local NUMA nodes if the system is running a sub-NUMA
+  clustering (SNC) mode. Without this options balloons would be
+  tightly packed on a single socket allowing the other CPU socket
+  sleep and save power. Here we have optimized for performance.
+- The list of `balloonTypes` includes two application-specific balloon
+  types: one for tgi and one for tei containers.
+- `matchExpressions` of a balloon type enable matching containers that
+  should be run in balloons of this type. We select tei and tgi
+  containers into their special balloon types based on container
+  name. Matching could be done based on labels and pod name, too.
+- `preferNewBalloon: true` on both tei and tgi balloon types means
+  that when a container is assigned into this balloon type and it is
+  possible to create a new balloon of this type because there are
+  enough free CPUs in the system, then the new balloon will be created
+  for the container. As a result, both tei and tgi containers will get
+  dedicated set of CPUs, unlike other containers that will run in the
+  default balloon type. Each container is allowed to use only CPUs of
+  the balloon where they are assigned.
+- `minCPUs: 16` and `minCPUs: 32` define the minimum number of CPUs in
+  a balloon. Created balloon will never be smaller even if containers
+  assigned to a balloon of this type would request fewer or no CPUs at
+  all. Correspondingly `maxCPUs` could be used to set an upper limit
+  for CPUs.
+- `hideHyperthreads: true` means that containers in balloons of this
+  type are allowed to use only single CPU hyperthread from each full
+  CPU core in the balloon. By default, both using hyperthreads of all
+  CPUs in the balloon is allowed. Note that when `true`, both
+  hyperthreads are allocated to the balloon in any case, preventing
+  alloating them into other balloons. This ensures that the whole CPU
+  core is dedicated to containers in these balloons only.
+
+For more information about the configuration and the balloons resource
+policy, refer to the balloons
+[documentation](https://containers.github.io/nri-plugins/stable/docs/resource-policy/policy/balloons.html).
+
+## NRI topology-aware resource policy
+
+NRI plugins include topology-aware resource policy, too. Unlike
```

Comment:
Looks like using topology aware resource policy is easier .. might be good to add a sentence or two stating why take on the extra effort of balloons such as greater control, performance improvements of x% above and beyond ..

Reply:
Added few sentences. However, did not give performance numbers compared to topology-aware policy, as they vary quite a bit based on resource requests/limits of containers. The topology-aware uses that information for coming up with cpusets. Currently our example pipeline does not include any container resource information, so the policy cannot do much.